### DISCUSSION TITLE
Automatically qualify column names

### DISCUSSION THREAD

**Discussion by gillchristian** [SeaQL/sea-query#681] [Ideas]
It happened a few times already that I forget to qualify column names in the select or where clauses of a query that contains a join and I end up with `column reference is ambiguous`.

```
Char::SizeW -> (Char::Table, Char::SizeW)
```

Is there a way to automatically qualify all the columns in a query by the enum's `::Table` field? And if not, is that a feature that could be considered? I'd like to give it a shot if maintainers are open to it.

**Comment by tyt2y3** [comment]
What implementation strategy are you thinking of?
I imagine, we can create a wrapper type `struct Wrapper<T: Iden>` that implements `IntoColumnRef` differently.

**Reply by Expurple** [reply]
I also want this feature, but for a different input type. I usually use tuples like `(my_table::Entity, my_table::Column::MyColumn)` with types autogenerated by `sea_orm`, and typing the table name twice is annoying. `ColumnTrait` objects already know the name of the table. I want to use them without `Entity`.

If I understand correctly, currently standalone `ColumnTrait` objects are converted `IntoColumnRef` using the common [impl for `T: IntoIden + 'static`](https://docs.rs/sea-query/0.30.5/sea_query/types/trait.IntoColumnRef.html#impl-IntoColumnRef-for-T) from `sea_query`. And the problem with implementing the desired behavior is that Rust won't allow to add a specialization. I don't really like the wrapper solution, people will just forget to call `my_table::Column::MyColumn.qualified()` (made up name for wrapping) like they forget to use a tuple. Maybe we can replace that impl with multiple impls for more specific `IntoIden` types? Is is possible to cover them all that way?

**Comment by Expurple** [comment]
Added this to https://github.com/SeaQL/sea-query/discussions/795 (in case the eventual solution will involve a breaking change).

Related thread on Discord: [Auto prefixing a column with its table name](https://discord.com/channels/873880840487206962/1382511513507729468)
